governance/src/agent.spec.js

```diff
@@ -0,0 +1,417 @@
+
+    // load the ABI from the specified file
+    // the call to getAbi will fail if the file does not exist
+    const abi = getAbi(abiFile);
+
+    // extract all of the event names from the ABI
+    const events = getObjectsFromAbi(abi, 'event');
+
+    // verify that at least the minimum list of supported events are present
+    MINIMUM_EVENT_LIST.forEach((eventName) => {
+      if (Object.keys(events).indexOf(eventName) === -1) {
+        throw new Error(`ABI does not contain minimum supported event: ${eventName}`);
+      }
+    });
+  });
+});
+
+const firstContractName = Object.keys(config.contracts)[0];
+const firstContract = config.contracts[firstContractName];
+const abi = getAbi(firstContract.abiFile);
+
+const invalidEvent = {
+  anonymous: false,
+  inputs: [
+    {
+      indexed: false,
+      internalType: 'uint256',
+      name: 'testValue',
+      type: 'uint256',
+    },
+  ],
+  name: 'TESTMockEvent',
+  type: 'event',
+};
+// push fake event to abi before creating the interface
+abi.push(invalidEvent);
+const iface = new ethers.utils.Interface(abi);
+
+
+// tests
+describe('monitor governance contracts for emitted events', () => {
+  describe('handleTransaction', () => {
+    let initializeData;
+    let handleTransaction;
+    let mockTxEvent;
+
+    // constants
+    const mockContractName = 'mockContractName';
+    const mockContractAddress = ethers.utils.hexZeroPad('0x1', 20);
+    const mockEventSignature = 'event mockEvent()';
+    const iface = new ethers.utils.Interface(abi);
+
+
+    // logs data for test case: address match + no topic match
+    const logsNoMatchEvent = [
+      {
+        address: mockContractAddress,
+        topics: [
+          ethers.constants.HashZero,
+        ],
+        data: '0x',
+      },
+    ];
+
+    // logs data for test case:  no address match + no topic match
+    const logsNoMatchAddress = [
+      {
+        address: ethers.constants.AddressZero,
+        topics: [
+          ethers.constants.HashZero,
+        ],
+        data: '0x',
+      },
+    ];
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      // initialize the handler
+      await (provideInitialize(initializeData))();
+      handleTransaction = provideHandleTransaction(initializeData);
+
+      validContractAddress = firstContract.address;
+
+
+      // initialize mock transaction event with default values
+      mockTxEvent = createTransactionEvent({
+        receipt: {
+          logs: [
+            {
+              name: '',
+              address: '',
+              signature: '',
+              topics: [],
+              data: `0x${'0'.repeat(1000)}`,
+              args: [],
+            },
+          ],
+        },
+      });
+    });
+
+    it('returns empty findings if contract address does not match', async () => {
+      // build txEvent
+      const txEvent = createTxEvent({
+        logs: logsNoMatchAddress,
+        addresses: { [ethers.constants.AddressZero]: true },
```

Comment:
I don't believe that this field needs to be populated.  It shouldn't have any part to play in the filtering process that the Bot performs.
```suggestion
```